glue/core/roi.py

```diff
@@ -1229,7 +1228,10 @@ def _sync_patch(self):
             self._patch.set_height(height)
             self._patch.set(**self.plot_opts)
             self._patch.set_visible(True)
+            self._axes.add_patch(self._patch)
         else:
+            if self._patch in self._axes.patches:
+                self._axes._remove_method(self._patch)
```

Comment:
I'm getting an error here the second time I try to use any particular ROI tool. I think we want to call `self._patch.remove()` (or I guess maybe `self._patch.remove_method()`, rather than the axes' remove method?

Reply:
Thanks, I have chosen the second option (first one seemed to let the tests pass as well, but I checked the KeyError on selecting an ROI a second time in the app was resolved by this one.
Still be good if you can confirm in your installation; I am not sure about the performance when using the app.

Reply:
This change has fixed the issue on my end